File: 1291a/main.py
def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

def calc(s):

    return sum([int(i) for i in s])

a = II()
for _ in range(a):
    b = II()
    s = I()
    while s and (int(s[-1]) & 1 == 0 or calc(s) & 1 == 1):
        s = s[:-1]

    if s == '':
        print(-1)
    else:
        print(s)

File: 1291a/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func calc(s string) int {
	sum := 0
	for _, v := range s {
		sum += int(v)
	}
	return sum
}

func solve(reader *bufio.Reader) {
	var b int
	var s string
	fmt.Fscan(reader, &b, &s)
	k := b - 1
	for len(s) > 0 && (int(s[k])&1 == 0 || calc(s)&1 == 1) {
		s = s[:k]
		k -= 1
	}
	if len(s) == 0 {
		fmt.Println(-1)
	} else {
		fmt.Println(s)
	}

}

func main() {
	reader := bufio.NewReader(os.Stdin)
	var a int
	fmt.Fscan(reader, &a)
	for i := 0; i < a; i++ {
		solve(reader)
	}
}
